### PR TITLE
Authentication api

### PR DESCRIPTION
**Pull request by emileswarts**
Create an ALB with a target group which scales containers on ec2 instances.
This is the new infrastructure for the Authentication API

### CODE REVIEW

File: govwifi-api/cluster.tf [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+resource "aws_ecs_task_definition" "authorisation-api-task" {
+  family = "authorisation-api-task-${var.Env-Name}"
+
+  container_definitions = <<EOF
+[
+    {
+      "volumesFrom": [],
+      "memory": 1900,
```
Comment (adrianclay) [inline govwifi-api/cluster.tf]: Where did this number come from?

Comment (RyanMacG) [reply govwifi-api/cluster.tf]: I'd assume from the existing clusters

Comment (RyanMacG) [reply govwifi-api/cluster.tf]: @adrianclay https://github.com/alphagov/govwifi-terraform/blob/master/govwifi-backend/cluster.tf#L20 lower than this limit

File: govwifi-api/cluster.tf [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          "name": "DB_USER",
+          "value": "${var.db-user}"
+        },{
+          "name": "DB_HOSTNAME",
+          "value": "db.${lower(var.aws-region-name)}.${var.Env-Subdomain}.service.gov.uk"
+        },{
+          "name": "ENVIRONMENT_NAME",
+          "value": "${var.Env-Name}"
+        },{
+          "name": "FRONTEND_API_KEY",
```
Comment (adrianclay) [inline govwifi-api/cluster.tf]: Does this get used?

**Comment by adrianclay** [comment]
Won't we require a `aws_appautoscaling_target` ?

https://www.terraform.io/docs/providers/aws/r/appautoscaling_target.html#ecs-service-autoscaling

File: govwifi-api/elb.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  load_balancer_type = "application"
+
+  tags {
+    Name = "api-alb-${var.Env-Name}"
+  }
+}
+
+resource "aws_alb_listener" "alb_listener" {
+  load_balancer_arn = "${aws_lb.api-alb.arn}"
+  port              = "8080"
```
Comment (adrianclay) [inline govwifi-api/elb.tf]: Doesn't 443 makes more sense for HTTPS?

File: govwifi-api/scaling-policy.tf [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+resource "aws_cloudwatch_metric_alarm" "cpualarm" {
+  count               = "${var.backend-cpualarm-count}"
+  alarm_name          = "${var.Env-Name}-api-cpu-alarm"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "2"
+  metric_name         = "CPUUtilization"
+  namespace           = "AWS/EC2"
+  period              = "60"
+  statistic           = "Average"
+  threshold           = "50"
```
Comment (adrianclay) [inline govwifi-api/scaling-policy.tf]: Be good to see this alarm + autoscaling tested and demonstrated as working on staging.

File: govwifi-api/cluster.tf (hunk 1 above)
Comment (emileswarts) [reply govwifi-api/cluster.tf]: This needs to be looked at when we performance test,  t2.medium has 4G ram, this is saying take just under half of it for the task.

**Pull request merged**